pkg/controller/pxc/controller.go

```diff
@@ -562,8 +562,8 @@ func (r *ReconcilePerconaXtraDBCluster) deploy(cr *api.PerconaXtraDBCluster) err
 	}
 
 	sslInternalHash, err := r.getSecretHash(cr, cr.Spec.PXC.SSLInternalSecretName, cr.Spec.AllowUnsafeConfig)
-	if err != nil && !k8serrors.IsNotFound(err) {
-		return fmt.Errorf("get secret hash error: %v", err)
+	if err != nil {
```

Comment:
```suggestion
	if err != nil && k8serrors.IsNotFound(err) && allowNonExistingSecret {
```

Reply:
we have exactly the same check inside getSecretHash func

Reply:
got it, thanks